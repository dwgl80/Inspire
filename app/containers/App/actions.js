import {
  GET_QUOTES,
  GET_QUOTES_SUCCESS,
  GET_QUOTES_FAILURE,
  SAVE_QUOTE,
  SAVE_QUOTE_SUCCESS,
  SAVE_QUOTE_FAILURE,
} from './constants';

const saveQuote = () => ({
  type: SAVE_QUOTE,
});

const saveSuccess = quote => ({
  type: SAVE_QUOTE_SUCCESS,
  quote,
});

const saveFailed = error => ({
  type: SAVE_QUOTE_FAILURE,
  error,
});

const getQuote = () => ({
  type: GET_QUOTES,
});

const quotesRetrieved = quotes => ({
  type: GET_QUOTES_SUCCESS,
  quotes,
});

const quotesFailed = error => ({
  type: GET_QUOTES_FAILURE,
  error,
});

export {
  getQuote,
  quotesRetrieved,
  quotesFailed,
  saveQuote,
  saveSuccess,
  saveFailed,
};
