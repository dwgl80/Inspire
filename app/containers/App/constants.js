/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const GET_QUOTES = 'inspire/App/GET_QUOTES';
export const GET_QUOTES_SUCCESS = 'inspire/App/GET_QUOTES_SUCCESS';
export const GET_QUOTES_FAILURE = 'inspire/App/GET_QUOTES_FAILURE';
export const SAVE_QUOTE = 'inspire/App/SAVE_QUOTE';
export const SAVE_QUOTE_SUCCESS = 'inspire/App/SAVE_QUOTE_SUCCESS';
export const SAVE_QUOTE_FAILURE = 'inspire/App/SAVE_QUOTE_FAILURE';
