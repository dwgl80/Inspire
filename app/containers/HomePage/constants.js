export const INPUT_QUOTE = 'inspire/HomePage/INPUT_QUOTE';
