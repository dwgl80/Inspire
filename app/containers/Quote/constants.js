export const CHANGE_LIKED = 'inspire/Quote/CHANGE_LIKED';
