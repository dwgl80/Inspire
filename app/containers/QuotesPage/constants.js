export const INPUT_SEARCH = 'inspire/QuotesPage/INPUT_SEARCH';
