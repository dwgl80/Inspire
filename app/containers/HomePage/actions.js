import { INPUT_QUOTE } from './constants';

const inputQuote = quote => ({
  type: INPUT_QUOTE,
  quote,
});

export { inputQuote };
