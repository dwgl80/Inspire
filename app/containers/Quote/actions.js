import { CHANGE_LIKED } from './constants';

const changeLiked = id => ({
  type: CHANGE_LIKED,
  id,
});

export { changeLiked };
