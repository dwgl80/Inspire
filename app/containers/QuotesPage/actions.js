import { INPUT_SEARCH } from './constants';

const inputSearch = query => ({
  type: INPUT_SEARCH,
  query,
});

export { inputSearch };
