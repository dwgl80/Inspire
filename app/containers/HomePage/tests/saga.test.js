import { put, takeEvery } from 'redux-saga/effects';

import { SAVE_QUOTE } from 'containers/App/constants';
import { saveSuccess, saveFailed } from 'containers/App/actions';

import saveAllQuotes, { saveQuotes } from '../saga';

const quote = 'shoot for the moon and the stars';

describe('saveQuotes Saga', () => {
  let saveQuotesGenerator;

  beforeEach(() => {
    saveQuotesGenerator = saveQuotes();

    const selectDescriptor = saveQuotesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = saveQuotesGenerator.next(quote).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the saveSuccess action if it saves the data successfully', () => {
    const response = quote;
    const putDescriptor = saveQuotesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(saveSuccess(response)));
  });

  it('should call the saveFailed action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = saveQuotesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(saveFailed(response)));
  });
});

describe('saveAllQuotes Saga', () => {
  const saveAllQuotesSaga = saveAllQuotes();

  it('should start task to watch for SAVE_QUOTE action', () => {
    const takeEveryDescriptor = saveAllQuotesSaga.next().value;
    expect(takeEveryDescriptor).toEqual(takeEvery(SAVE_QUOTE, saveQuotes));
  });
});
