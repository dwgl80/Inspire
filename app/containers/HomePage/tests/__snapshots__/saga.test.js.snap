// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saveQuotes Saga should call the saveFailed action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`saveQuotes Saga should call the saveFailed action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/saved/quotes",
      Object {
        "quote": "shoot for the moon and the stars",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`saveQuotes Saga should dispatch the saveSuccess action if it saves the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`saveQuotes Saga should dispatch the saveSuccess action if it saves the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/saved/quotes",
      Object {
        "quote": "shoot for the moon and the stars",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
